cvmfs/sync_mediator.cc

```diff
@@ -226,6 +226,8 @@ manifest::Manifest *SyncMediator::Commit() {
     return NULL;
   }
 
+  if (catalog_manager_->IsBalanced())
```

Comment:
What I mean is: if this is wrong, please add a test first that triggers the issue before fixing it.